stan/math/prim/meta/get.hpp

```diff
@@ -44,7 +44,7 @@ inline T get(const std::vector<T>& x, size_t n) {
  * @return n-th element of the \c Eigen \c Matrix or expression
  */
 template <typename T, typename = require_eigen_t<T>>
-inline auto get(const T& m, size_t n) {
+inline typename T::Scalar get(const T& m, size_t n) {
```

Comment:
Pretty sure this would work fine
```suggestion
inline scalar_type_t<T> get(const T& m, size_t n) {
```

Reply:
I prefere using member type instead of using metaprogram to keep the code simpler. But we can also go your way for consistency.

Reply:
done